src/components/InputWithStatus.tsx

```diff
@@ -0,0 +1,110 @@
+import { FieldStatus } from "@/types/Field";
+import CheckIcon from "@mui/icons-material/Check";
+import {
+  Box,
+  Divider,
+  IconButton,
+  InputBase,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import { useState } from "react";
+import { Controller, useFormContext, useWatch } from "react-hook-form";
+
+function InputWithStatus({
+  label,
+  placeholder,
+  name,
+  statusName,
+  className = "",
+}: {
+  label: string;
+  placeholder: string;
+  name: string;
+  statusName: string;
+  className?: string;
+}) {
+  const { control } = useFormContext();
+  const [isFocused, setIsFocused] = useState(false);
+
+  const statusValue = useWatch({
+    control,
+    name: statusName,
+  });
+
+  const toggleVerified = (
+    currentStatus: FieldStatus,
+    setStatus: (value: FieldStatus) => void,
+  ) => {
+    if (currentStatus !== FieldStatus.Error) {
+      setStatus(
+        currentStatus === FieldStatus.Verified
+          ? FieldStatus.Unverified
+          : FieldStatus.Verified,
+      );
+    }
+  };
+
+  const tooltipText =
+    statusValue === FieldStatus.Verified
+      ? "Mark as Unverified"
```

Comment:
No need to put this on a separate const something like this would be better:
 <Controller
        name={statusName}
        control={control}
        render={({ field: { value, onChange } }) => (
          <Tooltip title={ statusValue === FieldStatus.Verified ? "Mark as Unverified" : "Mark as Verified" } enterDelay={1000}>
            <IconButton
              onClick={() => toggleVerified(value, onChange)}
              data-testid={`toggle-status-btn-${statusName}`}
            >
              <CheckIcon
                className={
                  value === FieldStatus.Verified ? "text-green-500" : ""
                }
                data-testid={`status-icon-${statusName}`}
              />
            </IconButton>
          </Tooltip>
        )}
      />

Reply:
done